torch/csrc/jit/mobile/backport_manager.cpp

```diff
@@ -327,28 +227,162 @@ std::stringstream backport_v6_to_v5(std::stringstream& input_model_stream) {
   selective_copy(
       reader_bytecode, writer_bytecode, excluded_files, excluded_dirs);
 
-  update_bytecode_version(bytecode_values, kBytecodeVersionV5);
+  update_bytecode_version(bytecode_values, to_version);
   auto bytecode_tuple = c10::ivalue::Tuple::create(std::move(bytecode_values));
   SerializationStorageContext storage_context;
-  writeArchiveV5(
+  write_archive_v5_to_v7(
       writer_bytecode,
       c10::ivalue::Tuple::create(constants_values),
       /*archive_name=*/"constants",
       /*archive_dir=*/"",
       /*tensor_dir=*/"constants/",
       /*use_storage_context=*/true,
       storage_context);
-  writeArchiveV5(
+  write_archive_v5_to_v7(
       writer_bytecode,
       bytecode_tuple,
       /*archive_name=*/"bytecode",
       /*archive_dir=*/"",
       /*tensor_dir=*/"constants/",
       /*use_storage_context=*/true,
       storage_context);
+  return ouput_model_stream;
+}
+
+} // namespace
+
+/*
+ To add next backport function, for example, backport_vn_to_vn-1, create an
```

Comment:
I'm confused by the code organization: Why is this comment in the middle? There's also a bunch of opening and closing of namespace.

Can we instead use this structure:

1. The kBytecodeVersionVX constants
2. All the utility functions.
3. All the backport_vN_to_vN-1 functions.
4. BytecodeBackportFunction and BackportManager.

Right now it's hard to easily understand what are utility functions and what are the backport_vN_to_vN-1 entry points.

Also, rather than calling methods like write_archive_v5_to_v7, can we find better names and/or just document them?

Also, it seems wrong that you have what seem like general utilities (e.g. re_emit_model(...) ) which even take a version number but really don't work for all versions. In that case it would be best to break down those method into general utilities that the backport_vN_to_vN-1 functions call directly.
It may seem like we will have some repeated code but the general code will be so much more maintainable since you won't have thing like re_emit_model() that seems generic but it's not, nor functions like write_archive_v5_to_v7 that now are specialized to certain versions.

In summary:

* better error in smaller but generic utility functions at the cost of some duplicated code in backport_vN_to_vN-1.
* ideally, no version specific names/logic in the utilities --i.e. make them about their functionality, rather than tying them to the concept of a particular version.

That will also male backport_vN_to_vN-1 self documenting since we will be able to read at a high level what the backporting is doing, rather than traverse the current spaghetti code.